fuel-storage/src/impls.rs

```diff
@@ -201,7 +203,7 @@ impl<'a, T: StorageRead<Type>, Type: Mappable> StorageRef<'a, T, Type> {
         key: &<Type as Mappable>::Key,
         buf: &mut [u8],
     ) -> Result<Option<usize>, T::Error> {
-        self.0.read(key, buf)
+        self.0.read(key, 0, buf)
```

Comment:
We can specify offset here as well or add `read_with_offset`=)

Reply:
Done here: b2335f90a7d1a06ba81cfc4a4ba95c1417dcf851 
Is there any reason why `StorageRef` does not implement `StorageRead` btw? The only thing I see is that it does not implement `StorageSize`, which is a requirement for `StorageRead`.

Reply:
It doesn't implement any traits because it is used like `storage::<Mapable>().read` without importing the trait. This structure is used as a syntax sugar to simplify the work within the code